### PR TITLE
fixing dir not found

### PR DESCRIPTION
**Pull request by Aditya-eddy**
## Related Issue
  - closes [issue](https://github.com/keploy/keploy/issues/1750)

#### Describe the changes you've made
Changes made to check if keploy directory exists.
Attached screenshots in both cases:
Screenshot from 2024-03-29 09-48-38
Screenshot from 2024-03-29 09-44-15

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by Aditya-eddy** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Aditya-eddy** [comment]
recheck

**Comment by Aditya-eddy** [comment]
Hey @slayerjain intially I made changes in validated flags but reliased have to make change in replay.go(service/replay.go)
may not have done correctly. If i am wrong please correct me on this:
screenshot:
Screenshot from 2024-03-29 17-12-08

**Review by slayerjain — CHANGES_REQUESTED** [review]
Thanks for the PR. It would also be good to also check if the keploy folder exists but there are not testcases.

**Comment by slayerjain** [comment]
@Aditya-eddy I dont understand your concern. Can you please elaborate more?

**Comment by Aditya-eddy** [comment]
> It would also be good to also check if the keploy folder exists but there are not testcases.
Hey @slayerjain i have added to check the test-sets inside.
Screenshot from 2024-03-31 11-28-50

**Comment by Aditya-eddy** [comment]
- Hey @slayerjain any new changes in this?

File: pkg/service/replay/replay.go [hunk 1]
```diff
@@ -77,20 +77,37 @@ func (r *replayer) Start(ctx context.Context) error {
 		}
 	}()
 
-	// BootReplay will start the hooks and proxy and return the testRunID and appID
-	testRunID, appID, hookCancel, err := r.BootReplay(ctx)
+	absPath, err := filepath.Abs(r.config.Path)
```
Comment (charankamarapu) [inline pkg/service/replay/replay.go:80]: please do these checks in [validation of flags](https://github.com/keploy/keploy/blob/48ec8ba3ba8b5ea1389768ea0b3a559404d76e6e/cli/provider/cmd.go#L286)

Comment (charankamarapu) [inline pkg/service/replay/replay.go:80]: please do these checks in [validation of flags](https://github.com/keploy/keploy/blob/48ec8ba3ba8b5ea1389768ea0b3a559404d76e6e/cli/provider/cmd.go#L286)

**Review by charankamarapu — COMMENTED** [review]
Please address the comments

File: pkg/service/replay/replay.go (hunk 1 above)
Comment (Aditya-eddy) [reply pkg/service/replay/replay.go:80]: hey @charankamarapu do I have to both of the checks in the validation of flags:

For checking keploy folder
For checking if test-sets exists inside the keploy folder.

**Comment by Aditya-eddy** [comment]
Hey @charankamarapu any changes in this?

**Comment by charankamarapu** [comment]
will review by EOD

File: cli/provider/cmd.go [hunk 2]
```diff
@@ -387,7 +387,9 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 
 		c.cfg.Path = absPath + "/keploy"
 		if cmd.Name() == "test" {
+
 			testSets, err := cmd.Flags().GetStringSlice("testsets")
+			c.logger.Warn(fmt.Sprintf("%v", testSets))
```
Comment (charankamarapu) [inline cli/provider/cmd.go:392]: why is this needed..?

File: pkg/service/replay/replay.go [hunk 3]
```diff
@@ -89,21 +89,30 @@ func (r *Replayer) Start(ctx context.Context) error {
 			utils.LogError(r.logger, err, "failed to stop recording")
 		}
 	}()
+	if _, err := os.Stat(r.config.Path); os.IsNotExist(err) {
```
Comment (charankamarapu) [inline pkg/service/replay/replay.go:92]: Please add this validation in validateflags https://github.com/keploy/keploy/blob/cca2d422993deb6180a45d96dedf7fc3dba2e6a0/cli/provider/cmd.go#L260

File: pkg/service/replay/replay.go [hunk 4]
```diff
@@ -89,21 +89,30 @@ func (r *Replayer) Start(ctx context.Context) error {
...
+		errMsg := "keploy directory does not exist"
```
Comment (charankamarapu) [inline pkg/service/replay/replay.go:93]: Error should be "no test-sets found, please use record cmd to record test case" something like this.

**Review by charankamarapu — COMMENTED** [review]
Please address the comments

File: pkg/service/replay/replay.go (hunk 4 above)
Comment (Aditya-eddy) [reply pkg/service/replay/replay.go:93]: This is for checking if the keploy directory exists or not. We can change the message in errmsg down below.

Comment (charankamarapu) [reply pkg/service/replay/replay.go:93]: logging it as keploy directory, there is no use case for keploy directory. Test-set is a useCase where test cases are present, logging it as test set make more meaning.

**Comment by charankamarapu** [comment]
Hey @Aditya-eddy I think the issue is closed and resolved by another PR.
